contracts/modules/Experimental/Checkpoints/PLCRVotingCheckpoint.sol

```diff
@@ -0,0 +1,313 @@
+pragma solidity ^0.5.0;
+
+import "../../Module.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+contract PLCRVotingCheckpoint is Module {
+
+    using SafeMath for uint256;
+
+    enum Stage { PREP, COMMIT, REVEAL, RESOLVED }
+
+    struct Ballot {
+        uint256 quorum;       // Should be a multiple of 10 ** 16
+        uint256 totalSupply;
```

Comment:
We don't need to store `totalSupply`. We can just fetch the `totatSupply` at the checkpoint directly from token.